No class found for type implementing known interface

I've got the following (partial) schema:

```
type FilterSettings {
  id: ID!
  filterFields: [FilterField!]!
}

interface FilterType {
  id: ID!
  label: String!
}

type MultipleSelectFilterType implements FilterType {
  id: ID!
  label: String!
  options: [Option!]!
}

type Option {
  id: ID!
  label: String!
}
```

I've got a resolver that for this test does the following. This is the only place `FilterType` is returned.
```java
@Named
public class FilterFieldResolver implements GraphQLResolver<FilterField> {

  public List<FilterType> filterTypes(FilterField field) {
    MultipleSelectFilterType multipleSelectFilterType = new MultipleSelectFilterType("1", "1", "Cliëntmatch", null);
    return Arrays.asList(multipleSelectFilterType);
  }
}
```
It cannot instantiate `SchemaParser` because:
```
Caused by: com.coxautodev.graphql.tools.SchemaClassScannerError: Object type 'MultipleSelectFilterType' implements a known interface, but no class could be found for that type name.  Please pass a class for type 'MultipleSelectFilterType' in the parser's dictionary.
```
When I try to explicitly add the class to the dictionary using 
```java
@Bean
public SchemaDictionary dictionary() {
  return new SchemaDictionary().add(MultipleSelectFilterType.class);
}
``` 
it complains that the class already exists in the dictionary when starting up.

I've also tried adding dummy methods to query a `MultipleSelectFilterType` by id directly, but that doesn't help either. 

This effect can also been seen if you delete the `human` method in the schema in the example project:
```
type Query {
    # If episode omitted, returns the hero of the whole saga. If provided, returns the hero of that particular episode
    hero(episode: Episode): Character
#    # Find human by id
#    human(id: String!): Human
#    # Find droid by id
    droid(id: String!): Droid
    # Find character by id
    character(id: String!): Character
}
```
In that case it will give the same error about `Human`. When you uncomment the method it works again. But if I try this technique within my own project it still doesn't work.

My Java `MultipleSelectFilterType` class also actually implement the `FilterType` interface, and are named exactly the same as in the schema. Everything else seems to be working fine, so really no clue what I'm missing here.

How can I solve this problem?

--- a/src/main/kotlin/com/coxautodev/graphql/tools/SchemaClassScanner.kt
+++ b/src/main/kotlin/com/coxautodev/graphql/tools/SchemaClassScanner.kt
@@ -180,7 +180,7 @@ internal class SchemaClassScanner(initialDictionary: BiMap<String, Class<*>>, al
     private fun getAllObjectTypesImplementingDiscoveredInterfaces(): List<ObjectTypeDefinition> {
         return dictionary.keys.filterIsInstance<InterfaceTypeDefinition>().map { iface ->
             objectDefinitions.filter { obj -> obj.implements.filterIsInstance<TypeName>().any { it.name == iface.name } }
-        }.flatten().distinct()
+        }.flatten().distinctBy{ it.name }
     }
 
     private fun getAllObjectTypeMembersOfDiscoveredUnions(): List<ObjectTypeDefinition> {
@@ -192,7 +192,8 @@ internal class SchemaClassScanner(initialDictionary: BiMap<String, Class<*>>, al
 
     private fun handleInterfaceOrUnionSubTypes(types: List<ObjectTypeDefinition>, failureMessage: (ObjectTypeDefinition) -> String) {
         types.forEach { type ->
-            if(!unvalidatedTypes.contains(type) && !dictionary.containsKey(type)) {
+            val dictionaryContainsType = dictionary.filter{ it.key.name == type.name }.isNotEmpty()
+            if(!unvalidatedTypes.contains(type) && !dictionaryContainsType) {
                 val initialEntry = initialDictionary[type.name] ?: throw SchemaClassScannerError(failureMessage(type))
                 handleFoundType(type, initialEntry.get(), DictionaryReference())
             }

--- a/src/test/groovy/com/coxautodev/graphql/tools/SchemaClassScannerSpec.groovy
+++ b/src/test/groovy/com/coxautodev/graphql/tools/SchemaClassScannerSpec.groovy
@@ -250,7 +250,7 @@ class SchemaClassScannerSpec extends Specification {
         when:
         ScannedSchemaObjects objects = SchemaParser.newParser()
         // uncommenting the line below makes the test succeed
-//                .dictionary(InterfaceImplementation.NamedResourceImpl.class)
+                .dictionary(InterfaceImplementation.NamedResourceImpl.class)
                 .resolvers(new InterfaceImplementation())
                 .schemaString("""
                     type Query {

--- a/src/test/groovy/com/coxautodev/graphql/tools/SchemaClassScannerSpec.groovy
+++ b/src/test/groovy/com/coxautodev/graphql/tools/SchemaClassScannerSpec.groovy
@@ -246,6 +246,44 @@ class SchemaClassScannerSpec extends Specification {
         }
     }
 
+    def "scanner handles interface implementation that is not used as field type"() {
+        when:
+        ScannedSchemaObjects objects = SchemaParser.newParser()
+        // uncommenting the line below makes the test succeed
+//                .dictionary(InterfaceImplementation.NamedResourceImpl.class)
+                .resolvers(new InterfaceImplementation())
+                .schemaString("""
+                    type Query {
+                        query1: NamedResource
+                    }
+
+                    interface NamedResource {
+                        name: String!
+                    }
+
+                    type NamedResourceImpl implements NamedResource {
+                        name: String!
+                    }
+                """)
+                .scan()
+
+        then:
+        objects.definitions.findAll { it instanceof InterfaceTypeDefinition }.size() == 1
+    }
+
+    class InterfaceImplementation implements GraphQLQueryResolver {
+        NamedResource query1() { null }
+        NamedResourceImpl query2() { null }
+
+        static interface NamedResource {
+            String name()
+        }
+
+        static class NamedResourceImpl implements NamedResource {
+            String name() {}
+        }
+    }
+
     def "scanner handles custom scalars when matching input types"() {
         when:
             GraphQLScalarType customMap = new GraphQLScalarType('customMap', '', new Coercing<Map<String, Object>, Map<String, Object>>() {

